pkg/config/serving/server.go

```diff
@@ -20,23 +25,46 @@ func ToServerConfig(servingInfo configv1.HTTPServingInfo, authenticationConfig o
 	if err != nil {
 		return nil, err
 	}
+
 	if err := servingOptions.ApplyTo(&config.SecureServing, &config.LoopbackClientConfig); err != nil {
 		return nil, err
 	}
 
+	var lastApplyErr error
+
 	if !authenticationConfig.Disabled {
 		authenticationOptions := genericapiserveroptions.NewDelegatingAuthenticationOptions()
 		authenticationOptions.RemoteKubeConfigFile = kubeConfigFile
-		if err := authenticationOptions.ApplyTo(&config.Authentication, config.SecureServing, config.OpenAPIConfig); err != nil {
-			return nil, err
+
+		// In some cases the API server can return connection refused when getting the "extension-apiserver-authentication"
+		// config map.
+		err := wait.PollImmediateUntil(20*time.Second, func() (done bool, err error) {
+			lastApplyErr = authenticationOptions.ApplyTo(&config.Authentication, config.SecureServing, config.OpenAPIConfig)
```

Comment:
I'm not sure that wrapping `ApplyTo` into retry loop is a good idea here. Why not re-trying those operation that actually fail due to that error?

Reply:
@soltysh that goes deep into kube code, i'm not sure it will be accepted in upstream

Reply:
Gotcha, I started looking into the code to nail down those places and haven't found it easily, so that makes sense.